vm/systemSmartContracts/esdt.go

```diff
@@ -823,6 +918,78 @@ func (e *esdt) checkArgumentsForSpecialRoleChanges(args *vmcommon.ContractCallIn
 	return vmcommon.Ok
 }
 
+func isCreateNFTRoleInArgs(args [][]byte) bool {
+	for _, arg := range args {
+		if bytes.Equal(arg, []byte(core.ESDTRoleNFTCreate)) {
+			return true
+		}
+	}
+	return false
+}
+
+func checkIfCreateNFTRoleExists(args [][]byte, token *ESDTData) error {
+	if !isCreateNFTRoleInArgs(args) {
+		return nil
+	}
+
+	for _, esdtRole := range token.SpecialRoles {
```

Comment:
maybe you can create a function for this `checkIfRoleAlreadyExits` that return true or false

Reply:
would leave as it is.